sublime_linter.py

```diff
@@ -169,6 +169,15 @@ def on_hover(self, view, point, hover_zone):
             if persist.settings.get('show_hover_region_report'):
                 SublimeLinter.shared_plugin().open_tooltip(view, point, True)
 
+    def on_selection_modified_async(self, view):
+        if not view:
```

Comment:
Just curious. Is that really that Sublime calls us sometimes with no view? Just with nothing?

Reply:
I believe because the async-ness the view may no longer exist by the time this code is executed. It’s the reason behind a lot of the `if view` checks in this and other packages.

Reply:
Indeed, if the async thread was blocking on something earlier while the view was closed (which hopefully doesn't happen but could), then the view would most likely not exist anymore. However, I'm not sure if the function is even called at this point and, in addition, whether the view's ID might just be invalid instead. It's usually safe to check for `view.window()` for when a view object is invalid.

Reply:
Yeah okay, but you can't just `None` a var that is in our scope. python will not garbage collect this.

So I tested.

```
class UpdateState(sublime_plugin.EventListener):
    def on_selection_modified_async(self, view):
        current_pos = get_current_pos(view)
        import time
        time.sleep(1)
        print('test_view', view.id(), current_pos, view.file_name() or view.name(), view.window())


def get_current_pos(view):
    try:
        return view.rowcol(view.sel()[0].begin())
    except (AttributeError, IndexError):
        return -1, -1

```

If you run this, every callback blocks for a second. So you have plenty of time to close a view. Sublime will then still call you, even after the view is gone. The view is still valid, you can call `id()` and even `rowcol()`, but it is detached so `window()` returns `None`.